addons/mrp/models/mrp_production.py

```diff
@@ -1405,12 +1408,20 @@ def _get_quantity_produced_issues(self):
     def _action_generate_backorder_wizard(self, quantity_issues):
         ctx = self.env.context.copy()
         lines = []
+        ask_create_backorder = always_create_backorder = self.env['mrp.production']
         for order in quantity_issues:
-            lines.append((0, 0, {
-                'mrp_production_id': order.id,
-                'to_backorder': True
-            }))
-        ctx.update({'default_mrp_production_ids': self.ids, 'default_mrp_production_backorder_line_ids': lines})
+            if order.picking_type_id.create_backorder == 'ask' or (order.picking_type_id.create_backorder == 'never' and order.product_id.tracking == 'serial'):
+                lines.append((0, 0, {
+                    'mrp_production_id': order.id,
+                    'to_backorder': True
+                }))
+                ask_create_backorder += order
+            if order.picking_type_id.create_backorder == 'always':
+                always_create_backorder += order
+        (self - ask_create_backorder).with_context(ctx, skip_backorder=True, mo_ids_to_backorder=always_create_backorder.ids).button_mark_done()
```

Comment:
Keep the same behavior than current and inventory:
- Create 2 MO for 2 units each.
- First MO 2/2, second 1/2
- Validate both at the same time and discard the wizard.

The first MO should not be validated yet

Reply:
You can pass the valid in default_mrp_production_ids